pkg/chunk/grpc/index_client.go

```diff
@@ -0,0 +1,98 @@
+package grpc
+
+import (
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/util"
+)
+
+func (w *WriteBatch) Add(tableName, hashValue string, rangeValue []byte, value []byte) {
+	w.IndexEntry = append(w.IndexEntry, &IndexEntry{
+		TableName:  tableName,
+		HashValue:  hashValue,
+		RangeValue: rangeValue,
+		Value:      value,
+	})
+}
+
+func (w *WriteBatch) Delete(tableName, hashValue string, rangeValue []byte) {
+	// TODO: implement this to support deleting index entries for gRPC spec
```

Comment:
We should resolve this TODO.

Reply:
@pracucci I didn't notice any delete implementation for Cassandra and other stores. As it was TODO in respective store implementations. I reflected the same here. Do you want me to drop such comment with TODO?

Reply:
I think it has been added after you initiated this work. Please rebase `master`.